Makefile.gcc4

```diff
@@ -0,0 +1,33 @@
+# Makefile for ancient gcc4 (tested on Ubuntu 7.10 with 4.1.2-16ubuntu2)
+# Compiles Kaitai Struct C++ runtime and tests
+
+BUILD_DIR := build
+
+SOURCES := \
+	kaitai/kaitaistream.cpp \
+	tests/unittest.cpp
+
+OBJS := \
+	$(BUILD_DIR)/kaitaistream.o \
+	$(BUILD_DIR)/unittest.o
+
+DEFINES := -DKS_STR_ENCODING_ICONV -DGTEST_NANO
+
+LDFLAGS :=
+LDLIBS :=
+
+INCLUDES := -I.
+
+all: $(BUILD_DIR) $(BUILD_DIR)/unittest
```

Comment:
Yeah, and `all` should ideally be a [phony target](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html), but that's a trifle:

```suggestion
.PHONY: all
all: $(BUILD_DIR) $(BUILD_DIR)/unittest
```

Reply:
Cool, let me add that: f653068dc553f7eb452b26b5e71f2a861b065690